### DISCUSSION TITLE
Add an option to write files with escalated privileges (sudo)

### DISCUSSION THREAD

**Discussion by koibtw** [helix-editor/helix#11229] [General]
Add a command that will try to write files that current user doesn't have write permissions to with sudo or maybe add that functionality to the existing `:write`.

Similar to [vim-suda](https://github.com/lambdalisue/vim-suda).

https://github.com/user-attachments/assets/08166569-07fb-4be8-ba19-5280614903e8

**Comment by pascalkuthe** [comment]
you can use sudo-edit for this. This doesn't need to be builtin into the editor.

**Reply by koibtw** [reply]
yes i can but i think it's better and more comfortable to do it the editor way

**Reply by HernandoR** [reply]
Imagine you opened a /etc/app/config done a bunch of things, only to find out cannot bo saved. 


There's a useful command in vim that I wish helix could do the same function: w !sudo tee %

**Reply by CodingMadness** [reply]
this really should find its way in, it is an insane miserable experience when you did alot of changes to a file and then have to do text-yoga to somehow be able to save it...

**Comment by opensourcecheemsburgers** [comment]
This feels illegal, but adding this to your Helix config file should work...

```
[keys.normal]
A-w = '@:fmt<ret>%:| sudo tee %{buffer_name}<ret>:q!'
```

This macro is activated by pressing the `Alt` and `w` keys together. It does the following:
- `@` tells Helix the string is a macro.
- `:fmt` formats the document. This is necessary because Helix does not autoformat with this workaround.
- `<ret>` executes the previous `fmt` command for you by emulating an enter press.
- `%` selects the whole document.
- `:` opens the command interface.
- `|` pipes the selection to the command `tee` via stdin, tee reads the `stdin` and writes it to `%{buffer_name}`, which is a Helix variable that gives the relative (to where Helix was opened from) path of the active buffer.
- `<ret>` executes the aforementioned `tee` command for you. **IT WILL ASK YOU FOR SUDO PASSWORD.**
- `:q!` quits the document "without saving", but the file **has been written to**. 

The reasoning for the immediate exit is to salvage your modifications. This allows you to quickly reopen the file via `sudoedit`. Feel free to remove this part, but keep in mind that Helix does not realise you are writing to the file with this workaround.

**Comment by PitiBouchon** [comment]
That would be strange to have helix switcing user while editing a file.

Also, I do not have `sudo` so I'm not sure how would this be possible, and not have helix fail on some integrated (and no custom) commands
